src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -147,7 +147,69 @@ public RestAzureNS.AzureOperationResponse TriggerBackup()
 
         public RestAzureNS.AzureOperationResponse TriggerRestore()
         {
-            throw new NotImplementedException();
+            string vaultName = (string)ProviderData[VaultParams.VaultName];
+            string resourceGroupName = (string)ProviderData[VaultParams.ResourceGroupName];
+            CmdletModel.AzureFileShareRecoveryPoint recoveryPoint = ProviderData[RestoreBackupItemParams.RecoveryPoint]
+                as CmdletModel.AzureFileShareRecoveryPoint;
+            string storageAccountId = ProviderData[RestoreBackupItemParams.StorageAccountId].ToString();
+            string storageAccountLocation = ProviderData.ContainsKey(RestoreBackupItemParams.StorageAccountLocation) ?
+                ProviderData[RestoreBackupItemParams.StorageAccountLocation].ToString() : null;
+            string copyOptions = (string)ProviderData[RestoreFSBackupItemParams.ResolveConflict];
+            string sourceFilePath = ProviderData.ContainsKey(RestoreFSBackupItemParams.SourceFilePath) ?
```

Comment:
I remember Kartik telling that we need to pass Restore parameters as an object. Did you get any closure from him on that?

Reply:
Afaik, the parameter as objects concept was decided for upcoming workloads and we were going the usual way for afs